### PR TITLE
Allow adding services

### PR DESCRIPTION
**Pull request by wb9688**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
- [x] I agree to create a pull request for [NewPipe](https://github.com/TeamNewPipe/NewPipe) as soon as possible to make it compatible with the changed API.

### CODE REVIEW

File: extractor/src/main/java/org/schabi/newpipe/extractor/ServiceList.java [hunk 1]
```diff
@@ -44,20 +46,33 @@ private ServiceList() {
...
+    }
+
+    public static void addService(final Class<StreamingService> service) throws NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException {
+        services.add(service.getConstructor(new Class[]{int.class}).newInstance(nextService));
+        nextService++;
+    }
+
+    public static void replaceService(final Class<StreamingService> service, final int serviceId) throws NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException {
+        services.set(serviceId, service.getConstructor(new Class[]{int.class}).newInstance(serviceId));
+        nextService++;
```
Comment (TobiGr) [inline extractor/src/main/java/org/schabi/newpipe/extractor/ServiceList.java:76]: that's a security nightmare

**Comment by TobiGr** [comment]
If we want to allow loading additional services, we need a proper API management. e.g. the extractor needs an internal API version so that only compatible services are loaded.

File: extractor/src/main/java/org/schabi/newpipe/extractor/ServiceList.java (hunk 1 above)
Comment (wb9688) [reply extractor/src/main/java/org/schabi/newpipe/extractor/ServiceList.java:76]: Reflection in itself isn't a security nightmare. The NewPipe-side with `PathClassLoader` is perhaps not so great security wise (but other ways of having extensions can't be more secure, except when we add signing), but that's why I've clearly added a message in NewPipe that you should only install extensions that you trust, though perhaps I should emphasize that more.

**Comment by wb9688** [comment]
@TobiGr: I agree that we need an internal NewPipeExtractor version. For now I just added a simple version check in NewPipe, which is stupid. I also need to change NewPipe's DB scheme to store the service name instead of ID.

**Pull request closed without merging**